server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -118,14 +132,23 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
             return;
         }
 
-        String youtubeResBody = getYoutubeRes(API_KEY, accessToken.toString());
-        int numVideos = Integer.parseInt(req.getParameter("num_videos"));
+        String numVideosParam = req.getParameter("num_videos");
+        int numVideos = Integer.parseInt(DEFAULT_NUM_VIDS);
+        if (numVideosParam != null) {
+            numVideos = Integer.parseInt(numVideosParam);
+        }
+
+        String youtubeResBody = getYoutubeRes(API_KEY, accessToken.toString(), numVideos);
+        JsonObject youtubeJsonObj = JsonParser.parseString(youtubeResBody).getAsJsonObject();
 
         var genreCount = new HashMap<String, Integer>();
-        updateMusicCount(youtubeResBody, genreCount, numVideos);
+        updateMusicCount(youtubeJsonObj, genreCount);
+        int totalLiked = getTotalResults(youtubeJsonObj);
+        genreCount.put("totalLiked", totalLiked);
```

Comment:
So this hashmap will only have one key value pair?

Reply:
Assuming you plan to add more onto it

Reply:
In `updateMusicCount()` I add more key value pairs into the hash map 😃

Reply:
You should really have updateMusicCount return a new hashmap rather than declaring it here and having updateMusicCount modify its parameters. It tends to be bad practice to modify parameters as a way of communicating back to the parent function as it leads to confusion like the one in Seun's comment. I'd also rename the function to be `calculateGenreCount`.